### PR TITLE
Add note to development guide regarding GNU tools versions

### PR DESCRIPTION
**Pull request by ghost**
…especially on Mac OS X.

### CODE REVIEW

File: docs/devel/development.md [hunk 1]
```diff
@@ -50,6 +50,14 @@ Official releases are built using Docker containers. To build Kubernetes using
 Docker please follow [these
 instructions](http://releases.k8s.io/HEAD/build/README.md).
 
+### Local OS/shell environment
+
+Many of the Kubernetes development helper scripts rely on a fairly up-to-date GNU tools
```
Comment (mhrgoog) [inline docs/devel/development.md]: Would it be better to just have hack/verify-godeps.sh do a version check for diff? Then it's not a documentation issue but one that the script will find on its own and inform the user.

Comment (quinton-hoole) [inline docs/devel/development.md]: Yeah, I considered that approach too.  But there are so many cases of this sort of thing (tools where we implicitly rely on some 'recentish', dont-quite-know-which-version tool, and I don't even know a reliable way to check the versions.  Can perhaps try in a follow-up PR, but lets get this doc update in in the mean time.

Comment (mhrgoog) [inline docs/devel/development.md]: Then maybe reword "It requires <xxx> version. And note that Mac OS X may not ship with this by default."

Comment (quinton-hoole) [inline docs/devel/development.md]: What 'version' do you suggest using above?

**Comment by bgrant0607** [comment]
cc @luxas @dlorenc

**Comment by pwittrock** [comment]
@quinton-hoole Definitely an improvement.  Consider filing a 'help wanted' issue for automatically detecting compatibility of tools in build scripts.

**Comment by pwittrock** [comment]
@ghost I am inclined to 'lgtm' and improve incrementally as needed.  Do you have any other comments?
